subprojects/ear/src/main/java/org/gradle/plugins/ear/Ear.java

```diff
@@ -87,6 +87,8 @@ public Ear() {
         // this allows us to generate the deployment descriptor after recording all modules it contains
         CopySpecInternal metaInf = (CopySpecInternal) getMainSpec().addChild().into("META-INF");
         CopySpecInternal descriptorChild = metaInf.addChild();
+        // the generated descriptor should only be used if one does not already exist
+        descriptorChild.setDuplicatesStrategy(DuplicatesStrategy.EXCLUDE);
```

Comment:
❓ Why this isn't necessary when running without configuration caching?

Reply:
I could never pinpoint the exact difference in behavior between CC and non-CC scenarios that caused this bug.

What I clearly remember:
* with CC, we serialize a tree of copy specs using `DefaultCopySpec` for all nodes, whereas without CC, only the root spec is a `DefaultCopySpec`, whereas the child specs are `SingleParentCopySpec`s. 
* for a reason I could not single out, when walking the copy spec for the root, we would find the application.xml under two different copy specs: under the main spec, and again under the descriptor or metaInf spec, leading to duplicates. Without CC, it would only be found in the on of them (not sure if under the main spec, or the custom descriptor spec added by the Ear plugin.